pkg/config/config_template.yaml

```
@@ -1,857 +1,1238 @@
-# to disable logging to syslog entirely, please set 'log_to_console' to 'false' as well.
-# log_to_syslog: false
-#
-# If 'syslog_uri' is left undefined/empty, a local domain socket connection will be attempted
+## @param log_to_syslog - boolean - optional - default: false
+## Set to 'true' to enable logging to syslog.
+## Note: Even if this option is set to 'false', the service launcher of your environment
+## may redirect the Agent process' stdout/stderr to syslog. In that case, if you wish
+## to disable logging to syslog entirely, set 'log_to_console' to 'false' as well.
 #
-# syslog_uri:
+# log_to_syslog: false
+
+## @param syslog_uri - string - optional
+## Define a custom remote syslog uri if needed. If 'syslog_uri' is left undefined/empty,
+## a local domain socket connection is attempted.
 #
-# Set to 'true' to output in an RFC 5424-compliant format
+# syslog_uri: <SYSLOG_URI>
+
+## @param syslog_rfc - boolean - optional - default: false
+## Set to 'true' to output in an RFC 5424-compliant format for Agent logs.
 #
 # syslog_rfc: false
+
+## @param syslog_pem - string - optional
+## If TLS enabled, you must specify a path to a PEM certificate here.
 #
-# If TLS enabled, you must specify a path to a PEM certificate here
-#
-# syslog_pem: /path/to/certificate.pem
+# syslog_pem: <PEM_CERTIFICATE_PATH>
+
+## @param syslog_key - string - optional
+## If TLS enabled, you must specify a path to a private key here.
 #
-# If TLS enabled, you must specify a path to a private key here
+# syslog_key: <PEM_KEY_PATH>
+
+## @param syslog_tls_verify - boolean - optional - default: true
+## If TLS enabled, you may enforce TLS verification here.
 #
-# syslog_key: /path/to/key.pem
+# syslog_tls_verify: true
+
+{{ end -}}
+{{- if .Autoconfig }}
+
+##############################
+## Autoconfig Configuration ##
+##############################
+
+## @param autoconf_template_dir - string - optional - default: /datadog/check_configs
+## Directory containing configuration templates for Autoconfig.
 #
-# If TLS enabled, you may enforce TLS verification here (defaults to true)
+# autoconf_template_dir: /datadog/check_configs
+
+## @param config_providers - List of custom object - optional
+## The providers the Agent should call to collect checks configurations. Available providers are:
+##   * kubelet - The kubelet provider handles templates embedded in pod annotations.
+##   * docker -  The Docker provider handles templates embedded in container labels.
+##   * clusterchecks - The clustercheck provider retrieves cluster-level check configurations from the cluster-agent.
+##   * kube_services - The kube_services provider watches Kubernetes services for cluster-checks
+##
+## See https://docs.datadoghq.com/guides/autodiscovery/ to learn more
 #
-# syslog_tls_verify: true
+# config_providers:
+#  - name: kubelet
+#    polling: true
+#  - name: docker
+#    polling: true
+#  - name: clusterchecks
+#    grace_time_seconds: 60
+{{- if .ClusterChecks }}
```

When `.ClusterChecks` is false, this chunk expands to:
```
#  - name: clusterchecks
#    grace_time_seconds: 60#  - name: etcd
#    polling: true
```
(note the missing newline).

The only way I found to fix it is by adding an empty `{{ else }}` before the `{{ end -}}` 😕

---

ping @olivielpeau on this one.

---

Removing the leading `-` should re-introduce the newline that's missing, i.e.:

```suggestion
{{ if .ClusterChecks }}
```

(fyi, these `-` are there to remove newlines that are introduced before and after `{{ }}` blocks, if that makes sense)

---

Woha! 👏